app/lib/katello/resources/discovery/container_discovery.rb

```diff
@@ -0,0 +1,123 @@
+module Katello
+  class ContainerDiscovery < RepoDiscovery
+    attr_reader :found, :crawled, :to_follow
+    def initialize(url, crawled = [], found = [], to_follow = [],
+                   upstream_credentials_and_search = {
+                     upstream_username: nil,
+                     upstream_password: nil,
+                     search: '*'
```

Comment:
This will be `nil` if `upstream_credentials_and_search` is overridden by a caller. If you want it to default to `*` you can use `@search = upstream_credentials_and_search.fetch(:search, '*')`.